### ISSUE TITLE
[3D Model] Stage views are not described

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#869]
## Issue summary

When a 3D model is rotated, a screen reader will annouce the current stage view. Ex, "View from state right", "View from stage left." This does not help to _describe_ the stage view for someone who can't see the 3D model. Non-visual screen reader users should have equal access to content that sighted users have access to.

<details>
  <summary>Screenshots</summary>

  <img width="1051" alt="Dawn product page. A 3D model of an astronaut is highlighted. VoiceOver output contains the text, View from stage left." src="https://user-images.githubusercontent.com/1392632/141185093-0acc9378-123d-4da1-b7a3-2a1c801612ef.png">

</details>

#### HTML

```html
<model-viewer
  toggleable="true"
  src="…"
  camera-controls="true"
  style="--poster-color: transparent"
  data-shopify-feature="1.7"
  alt="{model.alt}"
  poster="…"
  data-js-focus-visible=""
  ar-status="not-presenting"
></model-viewer>
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. Navigate to a product with one or more 3D models available
3. Activate the 3D model activator control
4. With an active screen reader, use the virtual cursor to navigate to the 3D model and rotate using arrow keys

### Expected

- 3D model stages announcement to describe the models visuals and attributes

### Actual

- 3D model announces the stage angle only

## Recommendation

**This will be a multi-team effort.**

1. Allow the merchant to upload unique descriptions per stage view in the Theme editor as "alt text." (Shopify Theme Editor team)
2. Pass each stage description as string values to be accepted by the `model-viewer` component (Shopify Theme team/Google team)
3. Announce each stage description when the stage comes into view. (Google team)

With this, the screen reader would announce,
> _"View from stage {angle}, {stage.alt}"_

Review the post "[3D Model Accessibility – Provide user-defined descriptions per stage variant](https://scottvinkle.me/blogs/work/3d-model-accessibility#user-defined-descriptions-per-stage-variant)" for more details.

#### HTML

```html
<model-viewer
  toggleable="true"
  src="…"
  camera-controls="true"
  style="--poster-color: transparent"
  data-shopify-feature="1.7"
  alt="{model.alt}"
  alt-left="{model.alt.left}"
  alt-right="{model.alt.right}"
  alt-top="{model.alt.top}"
  …
  poster="…"
  data-js-focus-visible=""
  ar-status="not-presenting"
></model-viewer>
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.2.0
- **Component:** 3D Model
- **WCAG Principal:** Perceivable
- **WCAG SC:** [1.1.1 Non-text Content](https://www.w3.org/TR/UNDERSTANDING-WCAG20/text-equiv-all.html)
- **Severity:** High
- **Effort:** Very High
- **Related:** https://github.com/google/model-viewer/issues/442

**Comment by JonathanWade** [comment]
@elalish This is probably a bit of lift to get right and require coordination from us. Another route is to try and cram some of this into glTF metadata/XMP metadata, but thats another process. Love to discuss your thoughts on our next sync call.
